ipa-core/src/protocol/context/dzkp_field.rs

```diff
@@ -12,17 +25,332 @@ pub trait DZKPCompatibleField: Field {
+    /// This function is called by the prover.
+    fn convert_prover<'a>(
+        x_left: &'a BitArray32,
+        x_right: &'a BitArray32,
+        y_left: &'a BitArray32,
+        y_right: &'a BitArray32,
+        prss_right: &'a BitArray32,
+    ) -> impl Iterator<Item = UVPolynomials<Self>>;
+
+    /// This is similar to `convert_prover` except that it is called by the verifier to the left.
+    /// The verifier on the left uses its right shares, since they are consistent with the prover's left shares.
+    fn convert_verifier_left<'a>(
+        x_right: &'a BitArray32,
+        y_right: &'a BitArray32,
+        prss_right: &'a BitArray32,
+        z_right: &'a BitArray32,
+    ) -> impl Iterator<Item = SingleUVPolynomial<Self>>;
+
+    /// This is similar to `convert_prover` except that it is called by the verifier to the right.
+    /// The verifier on the right uses its left shares, since they are consistent with the prover's right shares.
+    fn convert_verifier_right<'a>(
+        x_left: &'a BitArray32,
+        y_left: &'a BitArray32,
+        prss_left: &'a BitArray32,
+    ) -> impl Iterator<Item = SingleUVPolynomial<Self>>;
 }
 
-// TODO(dm) - implement Basefield for Fp61BitPrime in follow up PR
+impl DZKPBaseField for Fp61BitPrime {
+    const INVERSE_OF_TWO: Self = Fp61BitPrime::const_truncate(1_152_921_504_606_846_976u64);
+    const MINUS_ONE_HALF: Self = Fp61BitPrime::const_neg(Self::INVERSE_OF_TWO);
+    const MINUS_TWO: Self = Fp61BitPrime::const_neg(Fp61BitPrime::const_truncate(2));
+
+    // Convert allows to convert individual bits from multiplication gates into dzkp compatible field elements
+    // We use the conversion logic from https://eprint.iacr.org/2023/909.pdf
+    //
+    // This function does not use any optimization.
+    //
+    // Prover and left can compute:
+    // g1=-2ac(1-2e),
+    // g2=c(1-2e),
+    // g3=a(1-2e),
+    // g4=-1/2(1-2e),
+    //
+    // Prover and right can compute:
+    // h1=bd(1-2f),
+    // h2=d(1-2f),
+    // h3=b(1-2f),
+    // h4=1-2f,
+    //
+    // where
+    // (a,b,c,d,f) = (xleft, yright, yleft, xright, prssright)
+    // and
+    // e = ab⊕cd⊕ f
+    fn convert_prover<'a>(
+        x_left: &'a BitArray32,
+        x_right: &'a BitArray32,
+        y_left: &'a BitArray32,
+        y_right: &'a BitArray32,
+        prss_right: &'a BitArray32,
+    ) -> impl Iterator<Item = UVPolynomials<Fp61BitPrime>> {
+        x_left
+            .chunks(RECURSION_CHUNK_SIZE)
+            .zip(y_right.chunks(RECURSION_CHUNK_SIZE))
+            .zip(y_left.chunks(RECURSION_CHUNK_SIZE))
+            .zip(x_right.chunks(RECURSION_CHUNK_SIZE))
+            .zip(prss_right.chunks(RECURSION_CHUNK_SIZE))
+            .map(|((((a, b), c), d), f)| {
+                // e = ab⊕cd⊕ f = x_left * y_right ⊕ y_left * x_right ⊕ prss_right
+                let e: BitArray<[u8; RECURSION_CHUNK_SIZE >> 3], Lsb0> = (BitArray::try_from(a)
```

Comment:
I do not understand why we are taking `RECURSION_CHUNK_SIZE` and shifting it up 3 bits. If all of a, b, c, d, and f are being iterated through in groups of `RECURSION_CHUNK_SIZE`, why is `e` initialised to 8x that size? Or do I just not understand the API for working with BitArray?

Reply:
shift by three is just the bit to byte conversion since BitArray wants to have the size in Bytes for the underlying storage container, Recursion_chunk_size is in bits. I will rename it in RECURSION_CHUNK_SIZE_BITS